common/docker-entrypoint.sh

```diff
@@ -85,56 +85,130 @@ if [ -n "$MAUTIC_CRON_DYNAMICS" ]; then
 echo >&2 "The following information will be prefilled into the installer (keep password field empty):"
 echo >&2 "Host Name: $MAUTIC_DB_HOST"
 echo >&2 "Database Name: $MAUTIC_DB_NAME"
 echo >&2 "Database Username: $MAUTIC_DB_USER"
 echo >&2 "Database Password: $MAUTIC_DB_PASSWORD"
+echo >&2 "========================================================================"
+echo >&2
+echo >&2
+
+# Write the provided configuration to the config so the installer prefills it
+echo >&2 "========================================================================"
+php /makeconfig.php
+echo >&2 "========================================================================"
+echo >&2
+echo >&2
+
+# Make sure our folder tree exists in our volume
+for path in spool cache logs media/files media/images; do
+  mkdir -p "/data/${path}"
+done
+
+# Make sure our web user owns everything in our volume
+chown -R www-data:www-data "/data"
+
+if ! grep -Fq "secret_key" /data/local.php; then
+  echo >&2 "========================================================================"
+  echo >&2 "Mautic not currently installed (no secret_key in local.php)"
+  echo >&2
 
-# Write the database connection to the config so the installer prefills it
-if ! [ -e app/config/local.php ]; then
-        php /makeconfig.php
+  if [ -n "$MAUTIC_URL" ] && [ -n "$MAUTIC_ADMIN_EMAIL" ] && [ -n "$MAUTIC_ADMIN_PASSWORD" ]; then
 
-        # Make sure our web user owns the config file if it exists
-        chown www-data:www-data app/config/local.php
-        mkdir -p /var/www/html/app/logs
-        chown www-data:www-data /var/www/html/app/logs
+    echo >&2 "URL & Admin credentials supplied, attempting automated mautic installation."
+    echo >&2
+
+    INSTALL_PARAMS=("$MAUTIC_URL")
+
+    if [ -n "$MAUTIC_INSTALL_FORCE" ]; then
+      INSTALL_PARAMS+=(-f)
+    else
+      if ! [[ $MAUTIC_URL =~ https://.* ]]; then
+        echo >&2 "URL not using HTTPS and MAUTIC_INSTALL_FORCE not provided.  please provide"
+        echo >&2 "a https url or set MAUTIC_INSTALL_FORCE true to continue."
+        echo >&2
+        echo >&2 "========================================================================"
+        sleep 5
+        exit 1
+      fi
+    fi
+
+    if [ -n "$MAUTIC_ADMIN_USERNAME" ]; then
+      INSTALL_PARAMS+=(--admin_username="$MAUTIC_ADMIN_USERNAME")
+    fi
+    if [ -n "$MAUTIC_ADMIN_FIRSTNAME" ]; then
+      INSTALL_PARAMS+=(--admin_firstname="$MAUTIC_ADMIN_FIRSTNAME")
+    fi
+    if [ -n "$MAUTIC_ADMIN_LASTNAME" ]; then
+      INSTALL_PARAMS+=(--admin_lastname="$MAUTIC_ADMIN_LASTNAME")
+    fi
+
+    sudo -Eu www-data php /var/www/html/bin/console mautic:install ${INSTALL_PARAMS[@]}
+  else
+    echo >&2 "URL & Admin credentials not supplied, please install mautic manually."
+  fi
+  echo >&2 "========================================================================"
+  echo >&2
+  echo >&2
+fi
+
+# clear mautic cache
+rm -rf /data/cache/*
```

Comment:
I found a couple of problems here with clearing cache with `rm -rf /data/cache/*` at the container entrypoint script:

- It deletes the Maxmind's IP database on startup (/data/cache/ip_data)
It would be better to to keep the IP database between restarts (for my use case)

- I can't even mount a volume for the IP database (/data/cache/ip_data) 
The container fails to start with the following error message: `rm: cannot remove '/data/cache/ip_data': Device or resource busy` if I mount a volume to persist the ip database

- I can't mount any volumes inside /data/cache/ - like /data/cache/prod
On servers with slow disk performance I mount a tmpfs volume for /data/cache/prod with the same error above.

A possible solution would be to add an environment variable (**MAUTIC_CLEAR_CACHE**) and only try to delete the cache if this environment variable exists.